utilities/pipelines/resourcePublish/Get-ModulesToUpdate.ps1

```diff
@@ -0,0 +1,373 @@
+
+0.3.630
+
+Generates a new version for the tables module.
+
+#>
+function Get-NewModuleVersion {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)]
+        [string] $TemplateFilePath
+    )
+
+    $Version = Get-ModuleVersionFromFile -TemplateFilePath $TemplateFilePath
+    $Patch = Get-GitDistance
+    $NewVersion = "$Version.$Patch"
+
+    $CurrentBranch = git symbolic-ref --short 'HEAD'
+    #$CurrentBranch = git branch --show-current
+    Write-Verbose "Current branch: [$CurrentBranch]" -Verbose
+    if (($CurrentBranch -ne 'main') -or ($CurrentBranch -ne 'master') ) {
+        $PreRelease = $CurrentBranch -replace '[^a-zA-Z0-9\.\-_]'
+        Write-Verbose "PreRelease: [$PreRelease]" -Verbose
+        $NewVersion = "$NewVersion-prerelease".ToLower()
+        #$NewVersion = "$NewVersion-prerelease-$PreRelease"
+        Write-Verbose "New version: [$NewVersion]" -Verbose
+    }
+
+    return $NewVersion
+}
+
+#endregion
+
+<#
+.SYNOPSIS
+Generates a hashtable with template file paths to update with a new version.
+
+.DESCRIPTION
+Generates a hashtable with template file paths to update with a new version.
+
+.PARAMETER TemplateFilePath
+Path to a deploy.bicep/json file.
+
+.EXAMPLE
+Get-ModulesToUpdate -TemplateFilePath 'C:\Repos\Azure\ResourceModules\arm\Microsoft.Storage\storageAccounts\tableServices\tables\deploy.bicep'
+
+
+#>#
+function Get-ModulesToUpdate {
+
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)]
+        [string] $TemplateFilePath
+    )
+
+    $ModuleFolderPath = Split-Path $TemplateFilePath -Parent
+    $TemplateFilesToUpdate = Get-TemplateFileToUpdate -ModuleFolderPath $ModuleFolderPath | Sort-Object FullName -Descending
+
+    $ModulesToUpdate = [System.Collections.ArrayList]@()
+    foreach ($TemplateFileToUpdate in $TemplateFilesToUpdate) {
+        $ModuleVersion = Get-NewModuleVersion -TemplateFilePath $TemplateFileToUpdate.FullName -Verbose
+        $ModulesToUpdate += @{
+            Version          = $ModuleVersion
+            TemplateFilePath = $TemplateFileToUpdate.FullName
+        }
+
+        $ParentTemplateFilesToUpdate = Get-ParentModuleTemplateFile -TemplateFilePath $TemplateFileToUpdate.FullName -Recurse
+        Write-Verbose "Found [$($ParentTemplateFilesToUpdate.count)] parent template files to update"
+        foreach ($ParentTemplateFileToUpdate in $ParentTemplateFilesToUpdate) {
+            $ParentModuleVersion = Get-NewModuleVersion -TemplateFilePath $ParentTemplateFileToUpdate.FullName
+
+            $ModulesToUpdate += @{
+                Version          = $ParentModuleVersion
+                TemplateFilePath = $ParentTemplateFileToUpdate.FullName
+            }
+        }
+    }
+
+    $ModulesToUpdate = $ModulesToUpdate | Sort-Object TemplateFilePath -Descending -Unique
+
+    Write-Verbose 'Update the following modules:'
+    $ModulesToUpdate | ForEach-Object {
+        Write-Verbose (' - [{0}] [{1}] ' -f $_.Version, $_.TemplateFilePath)
```

Comment:
Do you think it makes probably since to always print this for transparency reasons? Aka: Add a `-Verbose` to it?

Reply:
Fixed on all occurances